### PR TITLE
User and Role bug fixed

### PR DESCRIPTION
**Pull request by ajithandrewj**
## Description
The below issue is fixed
1 Insufficient Permissions for Admin User to Create Roles

## Type of Change
- [ yes ] Bug fix
- [ no ] New feature
- [ no ] Breaking change
- [ no ] Documentation update

## Checklist
- [ yes ] My code follows the style guidelines of this project
- [ yes ] I have performed a self-review of my own code
- [ yes ] I have commented my code, particularly in hard-to-understand areas
- [ no ] I have made corresponding changes to the documentation
- [ yes ] My changes generate no new warnings
- [ yes ] I have added tests that prove my fix is effective or that my feature works
- [ yes ] New and existing unit tests pass locally with my changes
- [ no ] Any dependent changes have been merged and published in downstream modules
- [ yes ] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [ yes ] Tasks are idempotent (can be run multiple times without changing state)
- [ no ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [ yes ] Playbooks are modular and reusable
- [ no ] Handlers are used for actions that need to run on change

## Documentation
- [ yes ] All options and parameters are documented clearly.
- [ yes ] Examples are provided and tested.
- [ yes ] Notes and limitations are clearly stated.

## Notes to Reviewers
The version based routing is done for user and role workflow module.

### CODE REVIEW

File: plugins/modules/user_role_workflow_manager.py [hunk 1]
```diff
@@ -1599,53 +1598,59 @@ def create_user(self, user_params):
...
+                    error_message = (
+                        "The Catalyst Center user '{0}' does not have the necessary permissions to 'create or update' a user through the API.".format(
+                            self.payload.get("dnac_username"))
+                    )
+                else:
+                    error_message = "Invalid email format for email '{0}' under username '{1}'".format(user_params.get("email"), user_params.get("username"))
+
+                return {"error_message": error_message}
+
+        error_message = "The specified version '{0}' does not have the 'add_user_api' functionality. Supported version(s) '2.3.5.3', '2.3.7.6'.".format(
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1651]: "The specified version '{0}' does not have the 'add_user_api' functionality. "
        "Supported version(s) from '2.3.5.3' onwards." 

Do you meant to say the supported versions from 2.3.5.3 ?

File: plugins/modules/user_role_workflow_manager.py [hunk 2]
```diff
@@ -1599,53 +1598,59 @@ def create_user(self, user_params):
...
+                self.log("Received API response from create_user: {0}".format(str(response)), "DEBUG")
+                self.created_user.append(user_params.get("username"))
+                return response
 
-            return {"error_message": error_message}
+            except Exception as e:
+                self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")
+                if "[403]" in str(e):
+                    error_message = (
+                        "The Catalyst Center user '{0}' does not have the necessary permissions to 'create or update' a user through the API.".format(
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1643]: "The Catalyst Center user '{0}' does not have the necessary permissions to 'create or update' a user via the API. "

File: plugins/modules/user_role_workflow_manager.py [hunk 3]
```diff
@@ -1599,53 +1598,59 @@ def create_user(self, user_params):
...
+                            self.payload.get("dnac_username"))
+                    )
+                else:
+                    error_message = "Invalid email format for email '{0}' under username '{1}'".format(user_params.get("email"), user_params.get("username"))
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1647]: error_message = "Invalid email format for '{0}' associated with username '{1}'.".format(
    user_params.get("email"), user_params.get("username")
)

File: plugins/module_utils/dnac.py [hunk 4]
```diff
@@ -117,14 +99,36 @@ def __init__(self, module):
...
+            elif v1 < v2:
+                return -1
+
+        # If versions are of unequal lengths, check remaining parts
+        if len(v1_parts) > len(v2_parts):
+            return 1 if any(part > 0 for part in v1_parts[len(v2_parts):]) else 0
+        elif len(v2_parts) > len(v1_parts):
+            return -1 if any(part > 0 for part in v2_parts[len(v1_parts):]) else 0
 
-    def get_ccc_version_as_integer(self):
```
Comment (madhansansel) [inline plugins/module_utils/dnac.py:123]: is not a problem.. Don't we use this in other modules? Have you checked it?

File: plugins/modules/user_role_workflow_manager.py [hunk 5]
```diff
@@ -1661,23 +1666,29 @@ def create_role(self, role_params):
...
+                self.created_role.append(role_params.get("role"))
+                return response
+
+            except Exception as e:
+                self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")
+                error_message = "The Catalyst Center user '{0}' does not have the necessary permissions to 'create a role' through the API.".format(
+                    self.payload.get("dnac_username"))
+                return {"error_message": error_message}
+
+        error_message = "The specified version '{0}' does not have the 'add_role_api' functionality. Supported version(s) '2.3.7.6'.".format(
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1689]: "The specified version '{0}' does not have the 'add_role_api' functionality. "
    "Supported version(s) from '2.3.7.6' onwards."

**Review by madhansansel — COMMENTED** [review]
I asked you to write in main itself..You don't need to write for all the functionalities..

  if self.compare_dnac_versions(self.get_ccc_version(), "2.3.5.3") < 0:
        self.status = "failed"
        self.msg = "The specified version '{0}' does not have the 'get_roles_api' functionality. Supported version(s) from '2.3.5.3' onwards.".format(
        self.payload.get("dnac_version"))
        self.log(self.msg, "ERROR")
        self.check_return_status()

File: plugins/modules/user_role_workflow_manager.py (hunk 1 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:1651]: Yes Madhan, Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 2 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:1643]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 5 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:1689]: Updated the code accordingly.

File: plugins/module_utils/dnac.py (hunk 4 above)
Comment (ajithandrewj) [reply plugins/module_utils/dnac.py:123]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 2 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:1643]: Updated the code accordingly.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**